File: module_me/src/main/java/com/fly/tour/me/NewsTypeListActivity.java
package com.fly.tour.me;

import android.app.Activity;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.fly.tour.common.event.RequestCode;
import com.fly.tour.common.mvvm.BaseMvvmRefreshActivity;
import com.fly.tour.common.util.ObservableListUtil;
import com.fly.tour.common.view.CommonDialogFragment;
import com.fly.tour.me.adapter.NewsTypeShowBindAdapter;
import com.fly.tour.me.databinding.ActivityNewsTypeListBinding;
import com.fly.tour.me.mvvm.factory.MeViewModelFactory;
import com.fly.tour.me.mvvm.viewmodel.NewsTypeListViewModel;
import com.refresh.lib.DaisyRefreshLayout;

/**
 * Description: <NewsTypeListActivity><br>
 * Author:      mxdl<br>
 * Date:        2019/07/02<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public class NewsTypeListActivity extends BaseMvvmRefreshActivity<ActivityNewsTypeListBinding, NewsTypeListViewModel> {
    private NewsTypeShowBindAdapter mNewsTypeShowAdapter;
    public int onBindLayout() {
        return R.layout.activity_news_type_list;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_toolbar_add, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int i = item.getItemId();
        if (i == R.id.menu_add) {
            startActivityForResult(new Intent(this, NewsTypeAddActivity.class), RequestCode.Me.NEWS_TYPE_ADD);
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void initView() {
        mNewsTypeShowAdapter = new NewsTypeShowBindAdapter(this, mViewModel.getList());
        mViewModel.getList().addOnListChangedCallback(ObservableListUtil.getListChangedCallback(mNewsTypeShowAdapter));
        mBinding.recview.setAdapter(mNewsTypeShowAdapter);
    }

    @Override
    public void initListener() {
        mNewsTypeShowAdapter.setDeleteClickLisenter(new NewsTypeShowBindAdapter.DeleteClickLisenter() {
            @Override
            public void onClickDeleteListener(final int id) {
                new CommonDialogFragment.Builder().setTitle("信息提示").setDescribe("确定删除吗？").setLeftbtn("取消").setRightbtn("确定").setOnDialogClickListener(new CommonDialogFragment.OnDialogClickListener() {
                    @Override
                    public void onLeftBtnClick(View view) {
                    }

                    @Override
                    public void onRightBtnClick(View view) {
                        mViewModel.deleteNewsTypeById(id);
                    }
                }).build().show(getSupportFragmentManager(), "dialog");
            }
        });
    }

    @Override
    public void initData() {
        mViewModel.refreshData();
    }


    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        switch (requestCode) {
            case RequestCode.Me.NEWS_TYPE_ADD:
                if (resultCode == Activity.RESULT_OK) {
                    autoLoadData();
                }
                break;
        }
    }

    @Override
    public Class<NewsTypeListViewModel> onBindViewModel() {
        return NewsTypeListViewModel.class;
    }

    @Override
    public ViewModelProvider.Factory onBindViewModelFactory() {
        return MeViewModelFactory.getInstance(getApplication());
    }

    @Override
    public void initViewObservable() {

    }

    @Override
    public int onBindVariableId() {
        return BR.viewModel;
    }

    @Override
    public DaisyRefreshLayout getRefreshLayout() {
        return mBinding.refviewNewsType;
    }
}

File: module_me/src/main/java/com/fly/tour/me/NewsTypeListActivity.kt
package com.fly.tour.me

import android.content.Context
import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.fly.tour.common.base.BaseRefreshActivity
import com.fly.tour.common.event.RequestCode
import com.fly.tour.common.view.CommonDialogFragment
import com.fly.tour.db.entity.NewsType
import com.fly.tour.me.adapter.NewsTypeListAdapter
import com.fly.tour.me.adapter.NewsTypeListAdapter.DeleteClickListener
import com.fly.tour.me.contract.NewsTypeListContract
import com.fly.tour.me.model.NewsTypeListModel
import com.fly.tour.me.presenter.NewsTypeListPresenter
import kotlinx.android.synthetic.main.activity_news_type_list.*
/**
 * Description: <NewsTypeListActivity><br>
 * Author:      mxdl<br>
 * Date:        2020/2/17<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
class NewsTypeListActivity :
    BaseRefreshActivity<NewsTypeListModel, NewsTypeListContract.View<NewsType>, NewsTypeListPresenter, NewsType>(),
    NewsTypeListContract.View<NewsType> {
    private var mNewsTypeListAdapter: NewsTypeListAdapter? = null
    override fun onBindRreshLayout(): Int {
        return R.id.mRefViewNewsType
    }

    override fun initPresenter(): NewsTypeListPresenter {
        return NewsTypeListPresenter(this)
    }

    override fun onBindLayout(): Int {
        return R.layout.activity_news_type_list
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_toolbar_add, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        var i = item.itemId
        if (i == R.id.menu_add) {
            startActivityForResult(
                Intent(this, NewsTypeAddActivity::class.java),
                RequestCode.Me.NEWS_TYPE_ADD
            )
        }
        return super.onOptionsItemSelected(item)
    }

    override fun initListener() {
        mRecViewNewsType.setLayoutManager(LinearLayoutManager(this))
        mNewsTypeListAdapter = NewsTypeListAdapter(this)
        mRecViewNewsType.setAdapter(mNewsTypeListAdapter)
        mNewsTypeListAdapter?.setDeleteClickLisenter(object : DeleteClickListener {
            override fun onClickDeleteListener(id: Int) {
                CommonDialogFragment.Builder().setTitle("信息提示").setDescribe("确定删除吗？")
                    .setLeftbtn("取消").setRightbtn("确定")
                    .setOnDialogClickListener(object : CommonDialogFragment.OnDialogClickListener {
                        override fun onLeftBtnClick(view: View) {

                        }

                        override fun onRightBtnClick(view: View) {
                            mPresenter?.deleteNewsTypeById(id)
                        }
                    }).build().show(supportFragmentManager, "dialog")
            }
        })
    }

    override fun initData() {
        mPresenter?.refreshData()
    }

    override fun refreshData(data: List<NewsType>) {
        mNewsTypeListAdapter?.refresh(data)
    }


    override fun loadMoreData(data: List<NewsType>) {
    }


    override fun onRefreshEvent() {
        mPresenter?.refreshData()
    }

    override fun onLoadMoreEvent() {
        mPresenter?.loadMoreData()
    }

    override fun onAutoLoadEvent() {
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        when (requestCode) {
            RequestCode.Me.NEWS_TYPE_ADD -> mPresenter?.refreshData()
        }
    }

    companion object {
        fun startNewsTypeListActivity(context: Context) {
            context.startActivity(Intent(context, NewsTypeListActivity::class.java))
        }
    }
}
